src/lib/remote-place-cache.ts

```diff
@@ -24,52 +18,56 @@ export type RemotePlace = {
 };
 
 export default class RemotePlaceCache {
-  private static cache: RemotePlaceDatastore = {};
+  private static cache: NodeCache;
 
   public static async getPlacesWithType(chtApi: ChtApi, contactType: ContactType, hierarchyLevel: HierarchyConstraint)
     : Promise<RemotePlace[]> {
-    const domainStore = await RemotePlaceCache.getDomainStore(chtApi, contactType, hierarchyLevel);
-    return domainStore;
+    const { domain } = chtApi.chtSession.authInfo;
+    const placeType = hierarchyLevel.contact_type;
+    const cacheKey = this.getCacheKey(domain, placeType);
+  
+    let places = this.getCache().get<RemotePlace[]>(cacheKey);
+    if (!places) {
+      places = await this.fetchRemotePlaces(chtApi, contactType, hierarchyLevel);
+      this.getCache().set(cacheKey, places);
+    }
+    return places;
+  }
+
+  private static getCache(): NodeCache {
+    if (!this.cache) {
+      this.cache = new NodeCache({ 
+        stdTTL: 3600,
+        checkperiod: 120 
+      });
+    }
+    return this.cache;
+  }
+
+  private static getCacheKey(domain: string, placeType?: string): string {
```

Comment:
under what situation should the `placeType` be undefined? i think we should probably assert in that case, rather than supporting it